### ISSUE TITLE
Need to clear an old backup that failed

### ISSUE THREAD

**Issue by instantlinux** [CrunchyData/postgres-operator#2513]
## Overview

After about 9 months, a backup failed in a way that's slamming my system with CPU overload and hundreds of thousands of logs. Examining it more closely, it actually only worked for the first 60 days, and has failed ever since; the problem became evident about 36 hours ago when alerts about CPU load (for `kube-controller-manager` service) started going off.

## Environment

- Platform: `Kubernetes`
- Platform Version: `1.19.4`
- PGO Image Tag: `centos7-4.5.1`
- Postgres Version: `centos7-12.5-4.5.1`
- Storage: `hostpath`

### Expected

System should continue running backups, or provide an error message that will help me to clear out the failure.

### Actual

Pods show up like this:
```
$ kubectl get pod -n pgo --context=sudo -o wide
NAME                                            READY   STATUS    RESTARTS   AGE     IP             NODE               NOMINATED NODE   READINESS GATES
backrest-backup-dbgitlab-226t6                  0/1     Pending   0          4m10s   <none>         <none>             <none>           <none>
backrest-backup-dbgitlab-6pdxh                  0/1     Error     0          5m18s   10.244.6.37    mckinley.ci.net    <none>           <none>
backrest-backup-dbgitlab-bbsxv                  0/1     Error     0          7m58s   10.244.6.23    mckinley.ci.net    <none>           <none>
backrest-backup-dbgitlab-bkxq4                  0/1     Error     0          8m38s   10.244.6.22    mckinley.ci.net    <none>           <none>
backrest-backup-dbgitlab-f9xz5                  0/1     Error     0          8m58s   10.244.6.21    mckinley.ci.net    <none>           <none>
backrest-backup-dbgitlab-kh945                  0/1     Error     0          9m18s   10.244.6.20    mckinley.ci.net    <none>           <none>
backrest-backup-dbgitlab-mvths                  0/1     Error     0          9m45s   10.244.6.19    mckinley.ci.net    <none>           <none>
dbgitlab-7c5f467f85-kz9pm                       1/1     Running   1          24d     10.244.1.103   elbrus.ci.net      <none>           <none>
dbgitlab-backrest-shared-repo-5bbcdc7d4-dlm9g   1/1     Running   0          43m     10.244.4.81    aconcagua.ci.net   <none>           <none>
postgres-operator-7b94775688-82f9w              4/4     Running   9          103m    10.244.6.39    mckinley.ci.net    <none>           <none>
```
Note that in addition to the failing pods (which are triggering massive logging and CPU overload the last 36 hours), I also see frequent restarts of the postgres-operator itself.

The last backup that worked (in November 2020) shows up thus:
```
$ pgo show backup dbgitlab
Handling connection for 8443

cluster: dbgitlab
storage type: local

stanza: db
    status: ok
    cipher: none

    db (current)
        wal archive min/max (12-1)

        full backup: 20201028-161505F
            timestamp start/stop: 2020-10-28 09:15:05 -0700 PDT / 2020-10-28 09:15:59 -0700 PDT
            wal start/stop: 0000000A000000090000001C / 0000000A000000090000001D
            database size: 104.0MiB, backup size: 104.0MiB
            repository size: 11.0MiB, repository backup size: 11.0MiB
            backup reference list: 
  ...
        full backup: 20201126-161505F
            timestamp start/stop: 2020-11-26 08:15:05 -0800 PST / 2020-11-26 08:15:54 -0800 PST
            wal start/stop: 0000000F00000011000000D0 / 0000000F00000011000000D1
            database size: 105.8MiB, backup size: 105.8MiB
            repository size: 11.2MiB, repository backup size: 11.2MiB
            backup reference list: 
```
The schedule was blank; I re-created the schedule thus but there's still something in the system (that I can't find in anywhere in the `ConfigMap`s.
```
$ pgo create schedule dbgitlab --schedule="15 16 * * *" --schedule-type=pgbackrest \
  --pgbackrest-backup-type=full --schedule-opts="--repo1-retention-full=30"
$ pgo show schedule -n pgo dbgitlab
Handling connection for 8443
dbgitlab-pgbackrest-full:
	schedule: 15 16 * * *
	schedule-type: pgbackrest
	backup-type: full
```
## Logs

Logs explain what happened but not how I can fix the problem.
```
$ kubectl logs backrest-backup-dbgitlab-25d6v -n pgo --context=sudo
time="2021-06-25T01:31:17Z" level=info msg="backrest backup command requested"
time="2021-06-25T01:31:17Z" level=info msg="command to execute is [pgbackrest backup  --db-host=10.244.1.103 --db-path=/pgdata/dbgitlab]"
time="2021-06-25T01:31:17Z" level=info msg="command is pgbackrest backup  --db-host=10.244.1.103 --db-path=/pgdata/dbgitlab "
time="2021-06-25T01:31:22Z" level=error msg="command terminated with exit code 29"
time="2021-06-25T01:31:22Z" level=info msg="output=[]"
time="2021-06-25T01:31:22Z" level=info msg="stderr=[WARN: option 'repo1-retention-full' is not set for 'repo1-retention-full-type=count', the repository may run out of space\n      HINT: to retain full backups indefinitely (without warning), set option 'repo1-retention-full' to the maximum.\nWARN: invalid backup '20201127-232103F' cannot be added to current backups\nWARN: a timeline switch has occurred since the 20201126-161505F backup, enabling delta checksum\n      HINT: this is normal after restoring from backup or promoting a standby.\n
  ERROR: [029]: new backup label '20201126-161505F_20210625-013119I' is not later than latest backup label '20201127-232103F.manifest.gz'\n       HINT: has the timezone changed?\n       HINT: is there clock skew?\n]"
time="2021-06-25T01:31:22Z" level=error msg="command terminated with exit code 29"
```
Whatever is running is creating a label tied to the date 20201126, and it's running without `repo1-retention-full` so it bombs out with `exit code 29` -- but the setting is buried somewhere that I have not been able to find (in hours of looking).

**Comment by instantlinux** [comment]
I've been trying to stabilize the backups but your documentation at [Disaster Recovery doc](https://access.crunchydata.com/documentation/postgres-operator/4.5.1/pgo-client/common-tasks/#disaster-recovery-backups-restores) omits information that would help me distinguish between a "backup" and a "schedule", and lacks information about how to find and remove unwanted settings. I wind up with way more differential backups than I want (it's really slamming my server too hard, causing k8s to destabilize) and failures from the pods like this new one: "ERROR: [050]: unable to acquire lock on file '/tmp/pgbackrest/db-backup.lock'".

Basically, I cannot use the backup feature of this system until I understand it more. And backups should _never_ be difficult to understand.

**Comment by andrewlecuyer** [comment]
Since this issue involves backup functionality from an older version of Crunchy Postgres for Kubernetes, I am proceeding with closing this issue.

If you have any additional questions - or if you would like to discuss backup scenarios, processes, functionality, etc. with Crunchy Postgres for Kubernetes v5 - further please feel free to reach out on the [PostgreSQL Operator community mailing list](https://groups.google.com/a/crunchydata.com/forum/#!forum/postgres-operator/join).

And for more information about Crunchy Postgres for Kubernetes v5 (including guidance for upgrading from v4), please see the latest documentation:

https://access.crunchydata.com/documentation/postgres-operator/latest/

Thanks!
